pkg/cluster/ClusterService.go

```diff
@@ -339,10 +340,18 @@ func (impl *ClusterServiceImpl) Update(ctx context.Context, bean *ClusterBean, u
 	model.PrometheusEndpoint = bean.PrometheusUrl
 
 	if bean.PrometheusAuth != nil {
-		model.PUserName = bean.PrometheusAuth.UserName
-		model.PPassword = bean.PrometheusAuth.Password
-		model.PTlsClientCert = bean.PrometheusAuth.TlsClientCert
-		model.PTlsClientKey = bean.PrometheusAuth.TlsClientKey
+		if bean.PrometheusAuth.UserName != "" {
+			model.PUserName = bean.PrometheusAuth.UserName
```

Comment:
by default string is empty, then why are we implementing this check?

Reply:
The problem was that req is having empty strings which ideally doesn't make sense to update in DB if it's just empty string. The parent check above this was trying to do the same thing : `bean.PrometheusAuth != nil` but since it's an object, the check was failing and we were updating the DB with empty values.